### PR TITLE
Add docs on how best to use sphinx-lint with pre-commit

### LINKED ISSUE #76: Make sphinx-lint faster
We're running sphinx-lint as part of pre-commit at CPython, and it's a really useful check. However, it's by _far_ the slowest check if you run `pre-commit run --all-files` locally. It would be great if we could speed it up somehow! (Note: I haven't looked into this at all yet; I don't know if there are any speedups that would be easily doable.)

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #76

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
 - focus on finding errors that are **not** visible to sphinx-build.
 
 
+## Using sphinx-lint
+
+To use sphinx-lint, simply run:
```
Comment (hugovk) [inline README.md:26]: Don't say simply:
* https://www.knowledgeowl.com/blog/posts/dont-say-simply-jim-fisher/
* https://www.youtube.com/watch?v=gsT2BBWBVmM&ab_channel=WritetheDocs


```suggestion
To use Sphinx Lint, run:
```

File: README.md [hunk 2]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+
+```console
+$ sphinx-lint file_or_directory_1, file_or_directory_2, ...
+```
```
Comment (hugovk) [inline README.md:30]: If there's no output to show, this makes it easier to copy/paste without the dollar:

```suggestion
```sh
sphinx-lint file_or_directory_1, file_or_directory_2, ...
```
```

File: README.md [hunk 3]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+
+Sphinx-lint can also be used via [pre-commit](https://pre-commit.com).
```
Comment (hugovk) [inline README.md:32]:
```suggestion
Sphinx Lint can also be used via [pre-commit](https://pre-commit.com).
```

File: README.md [hunk 4]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
 - focus on finding errors that are **not** visible to sphinx-build.
 
 
+## Using sphinx-lint
+
+To use sphinx-lint, simply run:
+
+```console
+$ sphinx-lint file_or_directory_1, file_or_directory_2, ...
```
Comment (ezio-melotti) [inline README.md:29]:
```suggestion
$ sphinx-lint           # check all dirs and files
$ sphinx-lint file.rst  # check a single file
$ sphinx-lint docs      # check a directory
$ sphinx-lint -i venv   # ignore a file/directory
$ sphinx-lint -h        # for more options
```

File: README.md [hunk 5]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+    hooks:
+      - id: sphinx-lint
+        args: [-j1]
+        types: [rst]
+```
+
+In particular, note that the `-j1` flag is recommended for use with pre-commit.
+By default, sphinx-lint uses `multiprocessing` to lint multiple files simultaneously,
+but this interacts poorly with pre-commit, which also attempts to use multiprocessing,
+leading to resource contention. Adding the `-j1` flag tells sphinx-lint not to use
```
Comment (hugovk) [inline README.md:47]:
```suggestion
By default, Sphinx Lint uses `multiprocessing` to lint multiple files simultaneously,
but this interacts poorly with pre-commit, which also attempts to use multiprocessing,
leading to resource contention. Adding the `-j1` flag tells Sphinx Lint not to use
```

File: README.md (hunk 1 above)
Comment (ezio-melotti) [inline README.md:26]:
```suggestion
To use `sphinx-lint`, simply run:
```

File: README.md (hunk 3 above)
Comment (ezio-melotti) [inline README.md:32]:
```suggestion
`sphinx-lint` can also be used via [pre-commit](https://pre-commit.com).
```

File: README.md [hunk 6]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+multiprocessing itself, deferring to pre-commit on the best way to delegate resources
+across a given computer's available cores.
```
Comment (hugovk) [inline README.md:49]:
```suggestion
across available cores.
```

File: README.md (hunk 1 above)
Comment (AlexWaygood) [reply README.md:26]: Argh, you've pointed this out to me before, and I always forget! You're quite right!

File: README.md [hunk 7]
```diff
@@ -21,6 +21,34 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+
+Sphinx-lint can also be used via [pre-commit](https://pre-commit.com).
+We recommend using a configuration like this:
+
+```yaml
+  - repo: https://github.com/sphinx-contrib/sphinx-lint
+    rev: LATEST_SPHINXLINT_RELEASE_TAG
+    hooks:
+      - id: sphinx-lint
+        args: [-j1]
```
Comment (hugovk) [inline README.md:40]: More verbose and descriptive:
```suggestion
        args: [--jobs=1]
```

File: README.md [hunk 8]
```diff
@@ -21,6 +21,38 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
 - focus on finding errors that are **not** visible to sphinx-build.
 
 
+## Using sphinx-lint
```
Comment (ezio-melotti) [inline README.md:24]:
```suggestion
## Using Sphinx Lint
```

File: README.md [hunk 9]
```diff
@@ -21,6 +21,38 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
 - focus on finding errors that are **not** visible to sphinx-build.
 
 
+## Using Sphinx Lint
+
+To use Sphinx Lint, run:
+
+```console
```
Comment (ezio-melotti) [inline README.md:28]:
```suggestion
```sh
```

File: README.md [hunk 10]
```diff
@@ -21,6 +21,38 @@ CPython](https://github.com/python/cpython/blob/e0433c1e7/Doc/tools/rstlint.py).
...
+$ sphinx-lint           # check all dirs and files
+$ sphinx-lint file.rst  # check a single file
+$ sphinx-lint docs      # check a directory
+$ sphinx-lint -i venv   # ignore a file/directory
+$ sphinx-lint -h        # for more options
+```
```
Comment (ezio-melotti) [inline README.md:34]:
```suggestion
```sh
sphinx-lint           # check all dirs and files
sphinx-lint file.rst  # check a single file
sphinx-lint docs      # check a directory
sphinx-lint -i venv   # ignore a file/directory
sphinx-lint -h        # for more options
```
```

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
A thought: rather than requiring users set the flag in their pre-commit config, can we set it as a default in https://github.com/sphinx-contrib/sphinx-lint/blob/main/.pre-commit-hooks.yaml? 

Docs: https://pre-commit.com/index.html#hooks-args

**Comment by AlexWaygood** [comment]
> A thought: rather than requiring users set the flag in their pre-commit config, can we set it as a default in https://github.com/sphinx-contrib/sphinx-lint/blob/main/.pre-commit-hooks.yaml?
> 
> Docs: https://pre-commit.com/index.html#hooks-args

Oh, very good idea!

**Pull request merged**